src/modules/blocks/rsvp/attendee-registration/container.js

```diff
@@ -36,9 +36,47 @@ const mapStateToProps = ( state ) => ( {
 	hasAttendeeInfoFields: selectors.getRSVPHasAttendeeInfoFields( state ),
 	isCreated: selectors.getRSVPCreated( state ),
 	isDisabled: getIsDisabled( state ),
+	isModalOpen: selectors.getRSVPIsModalOpen( state ),
 } );
 
+const mapDispatchToProps = ( dispatch, ownProps ) => {
+	return {
+		onClick: () => {
+			dispatch( actions.setRSVPIsModalOpen( true ) );
+		},
+		onClose: () => {
+			dispatch( actions.setRSVPIsModalOpen( false ) );
+		},
+		onIframeLoad: ( iframeWindow ) => {
+			const handleUnload = () => {
+				// remove unload listener
+				removeUnloadListener( iframeWindow );
+
+				// check if there are meta fields
+				const metaFields = iframeWindow.document.querySelector( '#tribe-tickets-attendee-sortables' );
+				const hasFields = Boolean( metaFields.firstElementChild );
+
+				// dispatch actions
+				dispatch( actions.setRSVPHasAttendeeInfoFields( hasFields ) );
+				dispatch( actions.setRSVPIsModalOpen( false ) );
+			};
+
+			const removeUnloadListener = ( win ) => {
```

Comment:
Can you move this above `handleUnload`?

Reply:
Ah, I missed the `handleUnload` being used in remove. We should probably just put that `removeUnloadListener` within `handleUnload`, since it's not being reused anywhere

Reply:
So I can't do that either, since when I call `removeEventListener`, I need to reference the function that I added in `addEventListener`. You can't reference the constant that you're defining.

Reply:
@nealfennimore Let us know if this resolves your concerns for now

Reply:
Okay, let's put a TODO to come back to this later. No point in holding this up